rule/imports-blacklist.go

```diff
@@ -0,0 +1,74 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// ImportsBlacklistRule lints given else constructs.
+type ImportsBlacklistRule struct{}
+
+// Apply applies the rule to given file.
+func (r *ImportsBlacklistRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+	blacklist := make([]string, len(arguments))
+
+	for i, arg := range arguments {
+		argStr, ok := arg.(string)
+		if !ok {
+			panic(fmt.Sprintf("Invalid argument to the imports-blacklist rule. Expecting a string, got %T", arg))
+		}
+		argStr = strings.ToLower(strings.TrimSpace(argStr))
```

Comment:
For a case-sensitive file system, upper cases will matter.

Reply:
I've added this lately because I don't see any benefits to be case sensitive, but It can helps to be insensitive:
Imagine you have the package: `github.com/MySuperUser/pkg`

using my _limited_ memory I may add
```toml
[rule.imports-blacklist]
arguments = ["github.com/mysuperuser/pkg"]
```
to the configuration file which will result to the package not being blacklisted if we are case sensitive.

And as `github.com/mysupperuser/pkg` can't exist when `github.com/MySuperUser/pkg` exists I don't see how it can lead to situations where the bad package is blacklisted.

Reply:
In a case-sensitive file system, both packages may exist. Sometime back, there were issues caused by [this](https://github.com/golang/dep/issues/1010).

Reply:
Wasn't aware, thanks for sharing.
Fixed in # 35e4c10